design/user_data_incentives.md

```diff
@@ -0,0 +1,124 @@
+# User interaction with Obscuro 
+
+In a typical transparent blockchain, there are service providers like "Infura" who act as the interface to the blockchain from the point of view of most users.
+
+They can perform this service relatively cheap, because all data is visible to everyone, so they can employ efficient caching techniques. 
+They also personally gain from understanding the user queries, and from analysing the exposed data.
+
+As a result, the crypto users have become used to this as a free service, and take it for granted.
+
+The data that is queried by users is:
+
+1. Transaction receipts. Either queried after submitting a transaction to understand whether it was executed successfully and included in a block, 
+   or to retrieve the submitted transactions.
+2. RPC "eth_call" - which are used to query the status of the various smart contracts. For example, by far the most used: "ERC20.getBalance"   
+3. RPC "get_balance" - to query the balance of the native ETH for an account.
+4. Event query or subscription - used by dApps to update the UIs 
+
+
+## Chains with privacy
+
+It gets a bit more complicated when privacy is added to the mix, because the results of the user interactions are dependent on the requester.
+Only logic run inside an approved TEE is able to return a result, since it is the only place where that data is visible.
+
+This means that it is now much more difficult to create a cheap caching layer.
+Also, the service provider will not gain any insights from this service since all traffic and processing is encrypted.
+
+
+## Delegate to the nodes
+
+The usual rescue in the crypto space is to employ incentives.
+
+Obscuro is a decentralised network of nodes with different roles who already have their own incentives (Aggregators and Verifiers).
+
+The ideal scenario is to have a large and diverse community of verifiers node to make sure that the network functions correctly.
+
+Is is natural to assign to the verifier nodes the additional task of servicing user requests.
+
+### Incentives
+
+There has to be a reason for a node operator to provide this service to users. If there is no payoff, then most operators could just block the ports
+and not accept incoming connections.
+
+Here is a list and analyisis of various potential reasons for offering this service
+
+#### 1. App developer
+
+Someone developed a dApp and wants to support its users, since they are directly interested in the growth of the application. It could be a company, or a DAO.
+E.g.: A game developer would not want to reply to requests made by a DEX user.
```

Comment:
This is a strong incentive. Supporting your hard-won users with a bullet proof verifier is a small price to pay to ensure reliability. From a business risk POV, relying on third parties for a mission critical component is undesirable and having your own verifiers mitigates this risk.

Reply:
yeah. It's a good "business" option. 
It's just that there are technical complexities